tests/bundle/EventListener/ConditionallyRemoveVaryHeaderListenerTest.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace Ibexa\Tests\Bundle\HttpCache\EventListener;
+
+use EzSystems\PlatformHttpCacheBundle\EventListener\ConditionallyRemoveVaryHeaderListener;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Event\ResponseEvent;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+
+class ConditionallyRemoveVaryHeaderListenerTest extends TestCase
+{
+    /** @var EzSystems\PlatformHttpCacheBundle\EventListener\ConditionallyRemoveVaryHeaderListener */
+    public $conditionallyRemoveVaryHeaderListener;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+        $this->conditionallyRemoveVaryHeaderListener = new ConditionallyRemoveVaryHeaderListener(['testroute1', 'testroute2']);
+    }
+
+    public function onKernelResponseProvider()
+    {
+        return [
+            [
+                'varyHeaders' => ['Cookie', 'vtest'],
+                'expectedVaryHeaders' => ['vtest'],
+            ],
+            [
+                'varyHeaders' => ['Authorization'],
+                'expectedVaryHeaders' => [],
+            ],
+            [
+                'varyHeaders' => ['cookie'],
+                'expectedVaryHeaders' => [],
+            ],
+            [
+                'varyHeaders' => ['authorization'],
+                'expectedVaryHeaders' => [],
+            ],
+            [
+                'varyHeaders' => ['Cookie', 'vtest'],
+                'expectedVaryHeaders' => ['vtest'],
+            ],
+            [
+                'varyHeaders' => ['vtest', 'Cookie'],
+                'expectedVaryHeaders' => ['vtest'],
+            ],
+            [
+                'varyHeaders' => ['cookie', 'vtest'],
+                'expectedVaryHeaders' => ['vtest'],
+            ],
+            [
+                'varyHeaders' => ['cookie', 'vtest', 'vtest2', 'vtest3', 'authorization'],
+                'expectedVaryHeaders' => ['vtest', 'vtest2', 'vtest3'],
+            ],
+            [
+                'varyHeaders' => ['cookie', 'vtest', 'vtest2', 'vtest3', 'authorization', 'vtest4'],
+                'expectedVaryHeaders' => ['vtest', 'vtest2', 'vtest3', 'vtest4'],
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider onKernelResponseProvider
+     */
+    public function testOnKernelResponse($varyHeaders, $expectedVaryHeaders)
```

Comment:
Similarly, add proper annotations to inform phpstan what array types are expected.
```suggestion
    /**
     * @dataProvider onKernelResponseProvider
     *
     * @param string[] $varyHeaders
     * @param string[] $expectedVaryHeaders
     */
    public function testOnKernelResponse($varyHeaders, $expectedVaryHeaders)
```